Documentation: add Dagster integration (#9439)
airbytehq/airbyte: docs/operator-guides/using-dagster-integration.md

---
description: Start triggering Airbyte jobs with Dagster in minutes
---

# Using the Dagster Integration 

Airbyte is an official integration in the Dagster project. The Airbyte Integration allows you to trigger synchronization jobs in Airbyte, and this tutorial will walk through configuring your Dagster Ops to do so.

The Airbyte Task documentation on Dagster project can be found [here](https://docs.dagster.io/_apidocs/libraries/dagster-airbyte).

## 1. Set up the tools

First, make sure you have Docker installed. We'll be using the `docker-compose` command, so your install should contain `docker-compose`.

> ### **Start Airbyte**

Reviewer: ```suggestion
### Start Airbyte
```

If this is your first time using Airbyte, we suggest going through our [Basic Tutorial](https://github.com/airbytehq/airbyte/tree/e378d40236b6a34e1c1cb481c8952735ec687d88/docs/quickstart/getting-started.md). This tutorial will use the Connection set up in the basic tutorial.

> For the purposes of this tutorial, set your Connection's **sync frequency** to **manual**. Prefect will be responsible for manually triggering the Airbyte job.

Reviewer: ```suggestion
For the purposes of this tutorial, set your Connection's **sync frequency** to **manual**. Dagster will be responsible for manually triggering the Airbyte job.
```

> ### **Install Dagster**

Reviewer: ```suggestion
### Install Dagster
```

    {
        "host": {"env": "AIRBYTE_HOST"},
        "port": {"env": "AIRBYTE_PORT"},
    }
)
sync_foobar = airbyte_sync_op.configured({"connection_id": "your-connection-uuid"}, name="sync_foobar")

@job(resource_defs={"airbyte": my_airbyte_resource})
def my_simple_airbyte_job():
    sync_foobar()

```

The Airbyte Dagster Resource accepts the following parameters:

* `host`: The host URL to your Airbyte instance.
* `port`: The port value you have selected for your Airbyte instance.
* `use_https`: If your server use secure HTTP connection.
* `request_max_retries`: The maximum number of times requests to the Airbyte API should be retried before failing.
* `request_retry_delay`: Time in seconds to wait between each request retry.

The Airbyte Dagster Op accepts the following parameters:
* `connection_id`: The Connection UUID you want to trigger
* `poll_interval`: The time in seconds that will be waited between successive polls.
* `poll_timeout`: he maximum time that will waited before this operation is timed out.

After running the file, `dagster job execute -f airbyte_dagster.py ` this will trigger the job with Dagster.

## That's it!

> Don't be fooled by our simple example of only one Prefect Flow. Airbyte is a powerful data integration platform supporting many sources and destinations. The Airbyte Prefect Task means Airbyte can now be easily used with the Prefect ecosystem - give it a shot!

Reviewer: ```suggestion
Don't be fooled by our simple example of only one Dagster flow. Airbyte is a powerful data integration platform supporting many sources and destinations. The Airbyte Dagster integration means Airbyte can now be easily used within the Dagster ecosystem!
```
